### PR TITLE
frydan solution

### PR DESCRIPTION
**Pull request by Frydan**

### CODE REVIEW

File: exercises/e001/results/Frydan/b32gen.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+meg = parser.add_mutually_exclusive_group()
+parser.add_argument('-c', '--characters', type=int, default=32, metavar='', help='Number of characters to generate (default: 32)')
+meg.add_argument('-l', '--lowercase', action='store_false', help='Output all lowercase (default choice)')
+meg.add_argument('-u', '--uppercase', action='store_true', help='Output all uppercase')
+
+args = parser.parse_args()
+
+def getString(numOfCharacters, notLowercase):	# main function for generating base32 strings
+	buff = ''
+	for i in range(numOfCharacters):
```
Comment (kucharskim) [inline exercises/e001/results/Frydan/b32gen.py]:
```suggestion
	for _ in range(numOfCharacters):
```

am I seeing this right, that `i` is unused?

File: exercises/e001/results/Frydan/b32gen.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
+#!/bin/python3
+
+import argparse
+import random
+
+parser = argparse.ArgumentParser(description='Base32 (RFC 4648) random character string generator')
+meg = parser.add_mutually_exclusive_group()
+parser.add_argument('-c', '--characters', type=int, default=32, metavar='', help='Number of characters to generate (default: 32)')
```
Comment (kucharskim) [inline exercises/e001/results/Frydan/b32gen.py]: this should be an optional argument, in other words, not `cmd.py -c 12` but `cmd.py 12`

Comment (kucharskim) [reply exercises/e001/results/Frydan/b32gen.py]: while at it, see `argparse` documentation for `%(default)s`

**Review by kucharskim — CHANGES_REQUESTED** [review]

**Comment by kucharskim** [comment]
I would put comments above the lines, not at the end. Current way comments are placed it hard to follow between line of code and the comment.

File: exercises/e001/results/Frydan/b32gen.py (hunk 1 above)
Comment (Frydan) [reply exercises/e001/results/Frydan/b32gen.py]: Yes You are right, i left it out accidentally after testing the whole range of possible characters.

File: exercises/e001/results/Frydan/b32gen.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
+#!/bin/python3
+
+import argparse
+import random
+
+parser = argparse.ArgumentParser(description='Base32 (RFC 4648) random character string generator')
+meg = parser.add_mutually_exclusive_group()
+parser.add_argument('characters', type=int, default=32, nargs='?', help='Number of characters to generate (default: 32)')
```
Comment (kucharskim) [inline exercises/e001/results/Frydan/b32gen.py]:
```suggestion
parser.add_argument('characters', type=int, default=32, nargs='?', help='Number of characters to generate (default: 32)')
meg = parser.add_mutually_exclusive_group()
```

This is nitpicking, but I would swap lines around.

**Review by kucharskim — DISMISSED** [review]

File: exercises/e001/results/Frydan/b32gen.py (hunk 3 above)
Comment (Frydan) [reply exercises/e001/results/Frydan/b32gen.py]: Just to visually separate mutually exclusive group from other arguments or is there another reason for it?

Comment (kucharskim) [reply exercises/e001/results/Frydan/b32gen.py]: This is a bit of bike-shedding, so don't worry or think too much about it IMO. I would swap it around as I would put line after line, objects which I'm modifying together. Currently you have:

```
parser = argparse.A...
meg = parser.add_mu...
parser.add_argume...
meg.add_argume...
meg.add_ar...
```

I would personally wrote:
```
parser = argparse.A...
parser.add_argume...
meg = parser.add_mu...
meg.add_argume...
meg.add_ar...
```

**Review by kucharskim — APPROVED** [review]
:+1:
